### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by Rostammahabadi**
refactored and added tests

### CODE REVIEW

File: lib/game_team.rb [hunk 1]
```diff
@@ -133,53 +133,33 @@ def self.fewest_tackles(season_id)
...
-    #loop through the values (games) and set them equal to the average of goals
-    team_averaged_goals = grouped_team.map do |ids, games|
-      goals_per_game = games.map {|game| game.goals}
-      games = (goals_per_game.sum / goals_per_game.length.to_f)
-    end
-    total_goals_per_team= Hash[grouped_team.keys.zip(team_averaged_goals)]
-    total_goals_per_team.key(total_goals_per_team.values.min)
+  def self.worst_offense
+    offensive_hash = total_goals_per_team
+    offensive_hash.key(offensive_hash.values.min)
```
Comment (stellakunzang) [inline lib/game_team.rb:150]: So fresh and so clean!

File: test/game_team_test.rb [hunk 2]
```diff
@@ -195,4 +195,16 @@ def test_it_can_return_best_offense_team_number
...
+    assert_equal 4, GameTeam.most_goals_scored("28")
+  end
+
+  def test_fewest_goals_scored_by_team_id
+    assert_equal 1, GameTeam.fewest_goals_scored("17")
+  end
+
+  def test_find_by_team
+    assert_equal 2, GameTeam.find_by_team("10").count
+  end
```
Comment (stellakunzang) [inline test/game_team_test.rb:209]: TDD FTW

File: lib/game_team.rb [hunk 3]
```diff
@@ -133,53 +133,33 @@ def self.fewest_tackles(season_id)
...
-  def self.best_offense
-    #grouped by team_id with keys being the team_id and values is an array of games
+  def self.total_goals_per_team
     grouped_team = all.group_by{|game| game.team_id}
-    #loop through the values (games) and set them equal to the average of goals
-    team_averaged_goals = grouped_team.map do |ids, games|
-      goals_per_game = games.map {|game| game.goals}
-      games = (goals_per_game.sum / goals_per_game.length.to_f)
+    grouped_team.keys.each_with_object({}) do |team_id , hash|
+      hash[team_id] = (grouped_team[team_id].sum(&:goals) / grouped_team[team_id].length.to_f).round(2)
```
Comment (Rostammahabadi) [inline lib/game_team.rb:139]: each_with_object creates a hash and iterates through each item of grouped_team therefore giving me access to the hash without calling `Hash.new` also this enumerable returns the created hash

File: lib/game_team.rb [hunk 4]
```diff
@@ -133,53 +133,33 @@ def self.fewest_tackles(season_id)
...
     end
-    total_goals_per_team= Hash[grouped_team.keys.zip(team_averaged_goals)]
-    total_goals_per_team.key(total_goals_per_team.values.max)
   end
 
-  def self.worst_offense
+  def self.best_offense
+    offensive_hash = total_goals_per_team
+    offensive_hash.key(offensive_hash.values.max)
+  end
```
Comment (perryr16) [inline lib/game_team.rb:146]: I think this is very readable as a deliverable and a helper.

File: lib/game_team.rb [hunk 5]
```diff
@@ -133,53 +133,33 @@ def self.fewest_tackles(season_id)
     fewest_tackles.first
   end
 
-  def self.best_offense
-    #grouped by team_id with keys being the team_id and values is an array of games
+  def self.total_goals_per_team
     grouped_team = all.group_by{|game| game.team_id}
```
Comment (perryr16) [inline lib/game_team.rb:137]: this looks like a killer module

**Review by foleymichelle9 — APPROVED** [review]
Looks great! Good job cutting down those lines of code

File: lib/team.rb [hunk 6]
```diff
@@ -3,14 +3,10 @@
 class Team < Collection
 
   def self.find_team_info(id)
-    id = id
     selected_team = all.find{|team| team.team_id==id}
+    required_data = ["abbreviation","franchise_id","link","team_id","team_name"]
     team_info = {}
```
Comment (perryr16) [inline lib/team.rb:8]: saving lines, every day

**Review by perryr16 — COMMENTED** [review]
approve -rp

**Pull request merged**